src/Okta.Sdk.UnitTests/PayloadHandlerShould.cs

```diff
@@ -0,0 +1,78 @@
+﻿// <copyright file="PayloadHandlerShould.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Okta.Sdk.Internal;
+using Okta.Sdk.UnitTests.Internal;
+using Xunit;
+
+namespace Okta.Sdk.UnitTests
+{
+    public class PayloadHandlerShould
+    {
+        [Fact]
+        public void SetHttpMessageContent()
+        {
+            // the TestPayloadHandler has a ContentType of "foo"
+            var testPayloadHandler = new TestPayloadHandler();
+            // register the payload handler so that it is found when the ContentType of the request is set. The OktaClient will register payload handlers as appropriate.
+            PayloadHandler.TryRegister(testPayloadHandler);
+            // set the ContentType of the request to match the ContentType of the TestPayloadHandler so that the request uses the TestPayloadHandler
+            var testHttpRequest = new HttpRequest { ContentType = "foo" };
+
+            var httpRequestMessage = new HttpRequestMessage();
+            httpRequestMessage.Content.Should().BeNull();
+            testPayloadHandler.SetHttpRequestMessageContent(testHttpRequest, httpRequestMessage);
+            httpRequestMessage.Content.Should().NotBeNull();
```

Comment:
Should we also assert the expected content?

Reply:
I'll add additional tests to validate content.

Reply:
Added additional tests in this file.

Reply:
Is it possible to assert that the content-type header of the content is the one we are expecting? Something like this:

```
httpRequestMessage.Content.ContentType.Should().Be("application/pkix-cert")
```
Unless we have other tests that already cover this assert.